plugins/modules/vm_snapshot_attach_disk.py

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2023, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+from hypercore.plugins.module_utils.task_tag import TaskTag
+
+DOCUMENTATION = r"""
+module: vm_snapshot_attach_disk
+
+author:
+  - Ana Zobec (@anazobec)
+short_description: Attach a disk from a snapshot to a VM on HyperCore API.
+description:
+  - Use this module to attach a disk from a snapshot to a desired VM on HyperCore API.
+version_added: 1.2.0
+extends_documentation_fragment:
+  - scale_computing.hypercore.cluster_instance
+seealso:
+  - module: scale_computing.hypercore.vm_snapshot_info
+options:  # TODO: Change options!
+  vm_name:
+    type: str
+    description:
+      - Name of the VM we want to attach a VM snapshot disk to.
+    required: False
+  source_snapshot_uuid:
+    type: str
+    description:
+      - UUID of the snapshot we want to use on a VM.
+    required: False
+  snapshot_type:
+    type: str
+    choices: [ user, automated ]
+    description:
+      - Specify a type that the chosen snapshot has to be.
+      - C(snapshot_type=user) - snapshot was created manually.
+      - C(snapshot_type=automated) - snapshot was created automatically through a schedule.
+    required: True
+  source_disk_slot:
+    type: int
+    description:
+      - Specify a snapshot disk slot to use on a VM.
```

Comment:
This sounds like it is slot on destination VM. But it is not. Change to:
```
      - Specify a disk slot from source snapshot to identify source disk.
```
Or to something better :)